From c274633994716c01d93863c1647796f6810aa1a2 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 14 Mar 2019 03:55:55 -0700
Subject: [PATCH 0569/1851] dt-bindings: media: vpss-scaler: Make max width and
 height mandatory properties

commit 619e530d7114cbf601bfbc42c97bf0b5c9873082 from
https://github.com/Xilinx/linux-xlnx.git

This patch makes xlnx,max-width and xlnx,max-height as mandatory dt
properties. This breaks backward compatibility.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/media/xilinx/xlnx,v-vpss-scaler.txt    | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/Documentation/devicetree/bindings/media/xilinx/xlnx,v-vpss-scaler.txt b/Documentation/devicetree/bindings/media/xilinx/xlnx,v-vpss-scaler.txt
index 148c727b5416..c29b5f487dbf 100644
--- a/Documentation/devicetree/bindings/media/xilinx/xlnx,v-vpss-scaler.txt
+++ b/Documentation/devicetree/bindings/media/xilinx/xlnx,v-vpss-scaler.txt
@@ -25,6 +25,12 @@ Required properties:
 - reset-gpios: Specifier for a GPIO that assert for VPSS Scaler reset.
   This property is mandatory for the Scaler
 
+- xlnx,max-height: Maximum number of lines.
+  Valid range from 64 to 4320.
+
+- xlnx,max-width: Maximum number of pixels in a line.
+  Valid range from 64 to 8192.
+
 - ports: Video ports, using the DT bindings defined in ../video-interfaces.txt.
   The scaler has an input port (0) and an output port (1).
 
@@ -41,11 +47,6 @@ Required port properties:
 
 - xlnx,video-width: Video width as defined in video.txt
 
-Optional properties:
-
-- xlnx,max-height: Maximum number of lines.
-- xlnx,max-width: Maximum number of pixels in a line.
-
 Example:
 
 	scaler_1:scaler@a0000000 {
@@ -57,6 +58,8 @@ Example:
 		xlnx,num-vert-taps = <8>;
 		xlnx,pix-per-clk = <2>;
 		reset-gpios = <&gpio 87 1>;
+		xlnx,max-width = <3840>;
+		xlnx,max-height = <2160>;
 
 		ports {
 			#address-cells = <1>;
-- 
2.31.1

